src/mono/mono/metadata/icall-eventpipe.c

```diff
@@ -491,6 +522,101 @@ eventpipe_execute_rundown (
 	return TRUE;
 }
 
+static
+gboolean
+eventpipe_walk_managed_stack_for_thread_func (
+	MonoStackFrameInfo *frame,
+	MonoContext *ctx,
+	gpointer data)
+{
+	g_assert_checked (frame != NULL);
+	g_assert_checked (data != NULL);
+
```

Comment:
These could be normal asserts.

Reply:
Used checked asserts to not add additional code into release builds since this callback will be called quite frequently, so I would either keep checked asserts or remove them (since not getting a frame and data probably indicate a static change in underlying API or caller).